api-reference/30 Data Layer/Store/3 Methods/push(changes).md

```diff
@@ -9,21 +9,41 @@ Pushes data changes to the store and notifies the [DataSource](/api-reference/30
 Data changes to be pushed.
 
 ---
-There are three possible data change types:
+Each data change is an object that can have the following fields:
+
+- `type`: String        
+Data change type: *"insert"*, *"update"*, or *"remove"*.
+
+- `data`: Object     
+Changes that should be applied to the store's data.
+
+- `key`: any        
+The key of the data item being updated or removed.
+
+- `index`: Number       
+The position at which to display a new data item in a UI widget bound to the store. To display the new data item first, set the `index` to 0. To add it to the end of the current page, set the `index` to -1.
+
+    The `index` field is optional. If you do not specify it, the new data item is added to the end of the dataset. However, if data is grouped or split into pages, this item will not appear in the UI widget until data is reshaped. In this case, specify the `index` for the pushed item to appear immediately.
+
+    The `index` field is ignored if **reshapeOnPush** is enabled (see the note below).
+
+[note] The **DataSource** does not automatically sort, group, filter, or otherwise shape pushed data. For this reason, the **DataSource** and the UI widget bound to it may become out of sync. To prevent this, enable the [reshapeOnPush](/api-reference/30%20Data%20Layer/DataSource/1%20Configuration/reshapeOnPush.md '/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#reshapeOnPush') option. We also recommend specifying the [pushAggregationTimeout](/api-reference/30%20Data%20Layer/DataSource/1%20Configuration/pushAggregationTimeout.md '/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#pushAggregationTimeout') to reduce the number of updates and recalculations.
```

Comment:
```suggestion
[note] The **DataSource** does not automatically sort, group, filter, or otherwise shape pushed data. For this reason, the **DataSource** and the UI widget bound to it can be out of sync. To prevent this, enable the [reshapeOnPush](/api-reference/30%20Data%20Layer/DataSource/1%20Configuration/reshapeOnPush.md '/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#reshapeOnPush') option. We also recommend specifying the [pushAggregationTimeout](/api-reference/30%20Data%20Layer/DataSource/1%20Configuration/pushAggregationTimeout.md '/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#pushAggregationTimeout') to reduce the number of updates and recalculations.
```